stac_fastapi/core/stac_fastapi/core/core.py

```diff
@@ -2587,9 +2600,32 @@ async def post_all_collections(
                 break
             token = next_token
 
-        links = []
+        if catalog_path:
+            catalog_url = f"catalogs/{catalog_path}"
+        else:
+            catalog_url = ""
+
+        links = [
+            {
+                "rel": Relations.root.value,
+                "type": MimeTypes.json,
+                "href": urljoin(base_url, catalog_url),
+            },
+            {
+                "rel": Relations.parent.value,
+                "type": MimeTypes.json,
+                "href": urljoin(base_url, catalog_url),
+            },
+            {
+                "rel": Relations.self.value,
+                "type": MimeTypes.json,
+                "href": f"{urljoin(base_url, catalog_url)}/collections",
```

Comment:
This doesn't look right - it should be a URL pointing to the specific collection. The link in dev is correct, though (eg, https://dev.eodatahub.org.uk/api/catalogue/stac/catalogs/supported-datasets/ceda-stac-catalogue/collections/sentinel2_ard for the first result in https://dev.eodatahub.org.uk/api/catalogue/stac/catalogs/supported-datasets/ceda-stac-catalogue/collections).

Reply:
This is only used to generate the final links in the `/collections` response:
```
{
            "rel": "root",
            "type": "application/json",
            "href": "https://dev.eodatahub.org.uk/api/catalogue/stac/catalogs/supported-datasets/ceda-stac-catalogue"
        },
        {
            "rel": "parent",
            "type": "application/json",
            "href": "https://dev.eodatahub.org.uk/api/catalogue/stac/catalogs/supported-datasets/ceda-stac-catalogue"
        },
        {
            "rel": "self",
            "type": "application/json",
            "href": "https://dev.eodatahub.org.uk/api/catalogue/stac/catalogs/supported-datasets/ceda-stac-catalogue/collections"
        }
```

Reply:
OK, understood. The parent and self links are correct.

The self link has a small wrinkle at https://dev.eodatahub.org.uk/api/catalogue/stac/collections - it's https://dev.eodatahub.org.uk/api/catalogue/stac//collections with a double `/`.

Reply:
Updated